test/polygeist-opt/paralleltogpu3.mlir

```diff
@@ -0,0 +1,112 @@
+// RUN: polygeist-opt --lower-affine --canonicalize --convert-parallel-to-gpu1 --canonicalize %s | FileCheck %s
+
+module {
+  func.func private @use(%arg0: index)
+  func.func @f1() {
+    %mc1 = arith.constant 1 : index
+    %mc1024 = arith.constant 1024 : index
+    %err = "polygeist.gpu_wrapper"(%mc1024, %mc1, %mc1) ({
```

Comment:
Do we need the gpu_wrapper? 

The number of threads being done the same way make sense, but I don't see why the number of blocks changing (presuming the optimization is correct), matters?

Reply:
`gpu_wrapper` indicates that the child region must be run on the gpu - perhaps a better name is `gpu_region`, I intend it to be a more generic operation than just for the gpu->gpu compilation pipeline. It is used independently of the `gpu_block` and `gpu_thread` which are only there to prevent the rest of the pipeline from doing transformations which alter the parallel structure of the gpu kernel and to retain the information about which index is from which dimension of the grid or block through the pipeline.

If we don't want to preserve the structure we omit the emission of the gpu_{block,thread}, then the compiler simplifies the code inside and the parallel-to-gpu pass tries its best to transform the now generic code in the region to match the gpu kernel execution model. The three args to the gpu_wrapper are the block dims which are sometimes used to check what the original block dims were if we want to best match that (could be removed as now we have a way of preserving the structure).

Reply:
If it is just to indicate that it should run ont he GPU, why does it take arguments?